tests/ttnn/unit_tests/operations/test_pad.py

```diff
@@ -128,6 +128,107 @@ def run_pad_rm_sharded(device, n, c, h, w, padding, torch_padding, value, shard_
     assert_with_pcc(torch_output_tensor, tt_output_tensor, 0.9999)
 
 
+def to_torch_padding(padspec):
+    def flatten_to_tuple(padding):
+        return tuple(sum(padding, ()))
+
+    def ttnn_pad_spec_to_padding(padspec):
+        input_tensor_start = padspec["input_tensor_start"]
+        pad_to_shape = padspec["pad_to_shape"]
+        input_shape = padspec["input_shape"]
+
+        padding = []
+        for i in range(len(pad_to_shape)):
+            this_dim_padding = (input_tensor_start[i], pad_to_shape[i] - input_shape[i] - input_tensor_start[i])
+            padding.append(this_dim_padding)
+        return padding
+
+    torch_padding = flatten_to_tuple(reversed(ttnn_pad_spec_to_padding(padspec)))
+    return torch_padding
+
+
+@pytest.mark.parametrize(
+    "input_shape, pad_to_shape, input_tensor_start, pad_value, input_sharded_memory_config_args",
+    [
+        [
+            (1, 1, 1, 4),
+            (1, 1, 1, 16),
+            (0, 0, 0, 0),
+            3.0,
+            {"core_grid": ttnn.CoreGrid(x=1, y=1), "strategy": ttnn.ShardStrategy.HEIGHT},
+        ],
+        [
+            # a reduced version of esmal's test case for UNet
+            (1, 1, 4, 4),
+            (1, 1, 4, 16),
+            (0, 0, 0, 0),
+            3.0,
+            {"core_grid": ttnn.CoreGrid(x=1, y=1), "strategy": ttnn.ShardStrategy.HEIGHT},
+        ],
+        [
+            # esmal's test case for UNet
+            (1, 1, 2 * 1056 * 160, 4),
+            (1, 1, 2 * 1056 * 160, 16),
+            (0, 0, 0, 0),
+            0.0,
+            {"core_grid": ttnn.CoreGrid(x=8, y=8), "strategy": ttnn.ShardStrategy.HEIGHT},
+        ],
+        [
+            # borys's first test case
+            (1, 1, 12, 8),
+            (1, 1, 12, 64),
+            (0, 0, 0, 0),
+            3.0,
+            {"core_grid": ttnn.CoreGrid(x=2, y=6), "strategy": ttnn.ShardStrategy.HEIGHT},
+        ],
+        [
+            # minimized borys second test case
+            (1, 1, 2, 4),
+            (1, 1, 4, 8),
+            (0, 0, 0, 0),
+            3.0,
+            {"core_grid": ttnn.CoreGrid(x=1, y=2), "strategy": ttnn.ShardStrategy.HEIGHT},
+        ],
+        [
+            # borys's second test case
+            (1, 2, 3, 4),
+            (1, 2, 32, 32),
+            (0, 0, 0, 0),
+            3.0,
+            {"core_grid": ttnn.CoreGrid(x=1, y=6), "strategy": ttnn.ShardStrategy.HEIGHT},
+        ],
+    ],
+)
+def test_pad_rm_sharded_stickwise(
+    device, input_shape, pad_to_shape, input_tensor_start, pad_value, input_sharded_memory_config_args
+):
+    input_shard_memory_config = ttnn.create_sharded_memory_config(input_shape, **input_sharded_memory_config_args)
+
+    torch_input_tensor = torch.ones(input_shape, dtype=torch.float32)
+    ttnn_input_tensor = ttnn.from_torch(
+        torch_input_tensor, dtype=ttnn.float32, layout=ttnn.ROW_MAJOR_LAYOUT, device=device
```

Comment:
I assume from_torch accepts memory_config param too

Reply:
I meant to file a bug for this. If you pass the sharded memory config to from_torch instead of doing the sharding separately you get a tensor with the wrong data (it was something like all 0's but I don't recall exactly.)

Reply:
@TT-BrianLiu , sound a bit familiar, you probably mentioned something like that on a sync before

Reply:
@sjameelTT also knows a thing or two about to/from_torch brokenness.